### PR TITLE
 Fix a missing feature: @jsdoc/salty: support for taffyDB({key:{isUndefined: false}})

### PR DESCRIPTION
**Pull request by microaaron**
support for taffyDB({key:{isUndefined: false}})

| Q                | A                                                                |
| ---------------- | ---------------------------------------------------------------- |
| Bug fix?         | yes                                                           |
| New feature?     | no                                                           |
| Breaking change? | no                                                           |
| Deprecations?    | no                                                           |
| Tests added?     | no                                                           |
| Fixed issues     |#2050 |
| License          | Apache-2.0                                                       |

Fix the problem that taffyDB({key:{isUndefined: false}}) cannot be parsed.

### Input code

```js
var db = new Salty([{ a: 3 }, { a: 1, b: 'hello' }, { a: 7, b: 2 }]);
var items_b = db({ b: {isUndefined: false}}).get();
JSON.stringify(items_b);
```
### Behavior
'[{"a":1,"b":"hello"},{"a":7,"b":2}]'

### CODE REVIEW

**Comment by microaaron** [comment]
@hegemonic Excuse me, Could you please reply to this PR? Please please.

File: packages/jsdoc-salty/lib/salty.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ const IS_UNDEFINED = {
   isUndefined: true,
 };
 
+const IS_NOTUNDEFINED = {
```
Comment (hegemonic) [inline packages/jsdoc-salty/lib/salty.js:28]: IS_NOT_UNDEFINED

**Comment by hegemonic** [comment]
Please add tests.

Also, I'm willing to accept this change, but please note that the `README` for this package says that it "supports only the TaffyDB features that JSDoc has historically used." It also says the following:

> Salty probably won't gain any new features, ever. It exists solely to meet the requirements of JSDoc.
>
> The exception is if a JSDoc template uses TaffyDB features that aren't available in Salty.

Finally, please don't email me about this pull request again.

**Comment by microaaron** [comment]
A simple test template:

```
const { taffy } = require('@jsdoc/salty');
let db = taffy([{ a: 3 }, { a: 1, b: 'hello' }, { a: 7, b: 2 }]);
let items_b = db({ b: {isUndefined: false}}).get();
JSON.stringify(items_b);
console.log(items_b);
```

**Comment by hegemonic** [comment]
> Thanks. In what way do you need to add tests?

Please add a test to [`packages/jsdoc-salty/test/specs/lib/salty.js`](https://github.com/jsdoc/jsdoc/blob/main/packages/jsdoc-salty/test/specs/lib/salty.js), similar to [this existing test](https://github.com/jsdoc/jsdoc/blob/ba5028c2f6b6bd7ae56ae75a36ef3ec1f18ebbaa/packages/jsdoc-salty/test/specs/lib/salty.js#L165-L173). It probably makes sense to add the new test right after the existing text.

After that, in the top-level directory for the repo, run `npm run default`. It'll run the tests and check for lint errors.

**Comment by hegemonic** [comment]
Sorry, but I can't help with that issue. I can't reproduce it locally or on GitHub's CI.

**Pull request merged**